core-lib/TestSuite/Minitest.ns

```diff
@@ -755,12 +755,28 @@ OTHER DEALINGS IN THE SOFTWARE. *)
 
   public class AsyncTestContext = TestContext <: Value ()(
     assert: promise resolvedWith: expectedResolution = (
-      (* TODO: add timeout support *)
-      ^ promise
-          whenResolved: [:actualResolution |
-            assert: actualResolution equals: expectedResolution ]
-          onError: [:actualResolution |
-            failWithMessage: 'Promise broken with error: ' + actualResolution asString ]
+      | resolved assertPP|
+      assertPP:: actors createPromisePair.
+      resolved:: false.
+
+      actors after: 100 do: [
+        resolved ifFalse: [
+          resolved:: true.
+          assertPP error: #PromiseTimeout.
+        ]
+      ].
+
+      promise
+        whenResolved: [:actualResolution |
+          resolved ifFalse: [
+            resolved:: true.
+            assert: actualResolution equals: expectedResolution.
+            assertPP resolve: 1.
+          ].
+        ]
+        onError: [:actualResolution |
+          failWithMessage: 'Promise broken with error: ' + actualResolution asString ].
+      ^ assertPP promise.
```

Comment:
What are all theses changes about?
This looks problematic to me.
And it doesn't seem to be a self-contained and documented change either.
The time out is rather arbitrary, too.

If this is needed, or desired, could you please break this change out?
And would be good to have an `assert:resolvedWith:timeout:` which takes an explicit parameter. 
`assert:resolvedWith:` can then call it explicitly.

Reply:
The current `assert:resolvedWith:` ignores the case where the promise is not resolved. I'll remove the change from this PR and put it into a Separate one with a version for specifying timeout.

Reply:
Thank you.

Reply:
Just for an explanation: I didn't actually notice that this was in a separate commit. Missed that yesterday.